### PR TITLE
add doc for versioning

### PR DESCRIPTION
**Pull request by ran-huang**
### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.1 (TiDB 6.1 versions)
- [x] v6.0 (TiDB 6.0 versions)
- [x] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by ran-huang** [comment]
@Ivy-YinSu

File: versioning.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
+---
+title: TiDB 版本规则
+summary: 了解 TiDB 版本发布的规则。
+---
+
+# TiDB 版本规则
+
+> **重要：**
+>
+> 始终升级到当前系列的最新补丁版本。
```
Comment (qiancai) [inline versioning.md:10]:
```suggestion
> 建议始终升级到当前系列的最新补丁版本。
```

File: versioning.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+从 TiDB 6.0.0 开始，TiDB 引入一个新的发布系列 DMR，并与 LTS 版共同作为 TiDB 的两个发布系列：
```
Comment (qiancai) [inline versioning.md:12]:
```suggestion
从 TiDB 6.0.0 开始，TiDB 引入一个新的版本系列 DMR，并与 LTS 版共同作为 TiDB 的两个版本系列：
```

File: versioning.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+- 长期支持版本 (Long-Term Support Releases, LTS)
+- 开发里程碑版本 (Development Milestone Releases, DMR)
+
+TiDB 5.0.0 及其之前的版本规则，请查看[历史版本](#不再沿用的历史版本号)。
+
+## 版本命名
+
+TiDB 版本的命名方式为 `X.Y.Z`。
+
+TiDB 从第一个版本 1.0 开始，`X` 每年依次递增。`Y` 代表会引入新的功能和改进，从 0 开始依次递增。`X.Y` 代表一个发布系列。`Z` 代表补丁版本的编号，一个发布系列开始发版时 `Z` 默认为 0，补丁版本的 `Z` 从 1 开始依次递增。
```
Comment (qiancai) [inline versioning.md:23]:
```suggestion
- 从 TiDB 1.0 起，`X` 每年依次递增并引入新的功能和改进。
- `Y` 从 0 开始依次递增，Y 的递增代表引入新的功能和改进。`X.Y` 代表一个发布系列。
- `Z` 代表补丁版本的编号，一个发布系列开始发版时 `Z` 默认为 0，补丁版本的 `Z` 从 1 开始依次递增。
```

File: versioning.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+开发里程碑版命名方式为 `X.Y`，系列版本号使用一个新的系列，`Z` 默认为0，并添加后缀 `-DMR`.
+
+示例版本:
+
+- 6.0.0-DMR
+
+## 补丁版本
+
+补丁版本 (Patch Release) 是基于生命周期内的长期支持版本 (LTS) 按需发布该系列的补丁版本。补丁版本主要包含 bug 修复、安全漏洞修复，不会包含新的功能。
```
Comment (qiancai) [inline versioning.md:48]:
```suggestion
补丁版本 (Patch Release) 是在长期支持版本 (LTS) 生命周期内按需发布的补丁版本。补丁版本主要包含 bug 修复、安全漏洞修复，不会包含新的功能。
```

File: versioning.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+补丁版本命名方式为 `X.Y` 系列版本号与长期支持版本保持不变，补丁版本号 `Z` 从 1 开始依次递增。
```
Comment (qiancai) [inline versioning.md:50]:
```suggestion
补丁版本命名方式为 `X.Y.Z`。其中，系列版本号 `X.Y` 与对应的长期支持版本保持一致，补丁版本号 `Z` 从 1 开始依次递增。
```

File: versioning.md [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+
+示例版本：
+
+- 1.1 Beta
+- 2.1 Beta
+- 4.0.0-beta.1
+
+### 内部测试版本
+
+内部测试版本 (Alpha Releases) 是内部测试版，会引入新的功能和改进。Alpha 版是当前系列版本的最初版本。Alpha版会存在一些 Bug，提供给尝鲜用户，可以用于测试最新的功能。
```
Comment (qiancai) [inline versioning.md:94]:
```suggestion
内部测试版本 (Alpha Releases) 是内部测试版，会引入新的功能和改进。Alpha 版是当前系列版本的最初版本。Alpha 版可能存在一些 Bug，提供给尝鲜用户，可以用于测试最新的功能。
```

File: versioning.md [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+- 开发里程碑版本 (Development Milestone Releases, DMR)
+
+TiDB 5.0.0 及其之前的版本规则，请查看[历史版本](#不再沿用的历史版本号)。
+
+## 版本命名
+
+TiDB 版本的命名方式为 `X.Y.Z`。
+
+TiDB 从第一个版本 1.0 开始，`X` 每年依次递增。`Y` 代表会引入新的功能和改进，从 0 开始依次递增。`X.Y` 代表一个发布系列。`Z` 代表补丁版本的编号，一个发布系列开始发版时 `Z` 默认为 0，补丁版本的 `Z` 从 1 开始依次递增。
+
```
Comment (qiancai) [inline versioning.md:24]: 或许长期支持版本、开发里程碑版本、补丁版本用一个表格呈现也不错，第一列是版本定义，第二列是命名方式，第三列是示例版本。(这个只是建议，可以不采纳）

File: releases/versioning-test.md [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+
+- 从 TiDB 1.0 起，`X` 每年依次递增并引入新的功能和改进。
+- `Y` 从 0 开始依次递增，Y 的递增代表引入新的功能和改进。`X.Y` 代表一个版本系列。
+- `Z` 代表补丁版本的编号，一个版本系列第一次发版时 `Z` 默认为 0，补丁版本的 `Z` 从 1 开始依次递增。
+
+| 版本系列 | 定义 | 命名方式 | 示例版本 |
+|:--|:-----|:---|:--|
+| 长期支持版本 (Long-Term Support Releases, LTS) | 长期支持版本约每六个月发布一次，会引入新的功能和改进，也会包含缺陷修复和安全漏洞修复，并会按需在版本生命周期内发布基于当前版本的补丁版本 (Patch Release)。 | 长期支持版本命名方式为 `X.Y`，系列版本号使用一个新的系列，`Z` 默认为 0。 | 6.1.0<br/>5.4.0 |
+| 开发里程碑版本 (Development Milestone Releases, DMR) | 开发里程碑版本约每两个月发布一次。如遇 LTS 发版，DMR 版本发版时间顺延两个月。DMR 会引入新的功能和改进，也会包含缺陷修复和安全漏洞修复。但 TiDB 不提供基于当前 DMR 版本的补丁版本，如有相关 bug 会在后续系列版本中陆续修复。 | 开发里程碑版命名方式为 `X.Y`，系列版本号使用一个新的系列，`Z` 默认为0，并添加后缀 `-DMR`. | 6.0.0-DMR |
+| 补丁版本 (Patch Release) | 补丁版本是在长期支持版本 (LTS) 生命周期内按需发布的补丁版本。补丁版本主要包含 bug 修复、安全漏洞修复，不会包含新的功能。 | 补丁版本命名方式为 `X.Y.Z`。其中，系列版本号 `X.Y` 与对应的长期支持版本保持一致，补丁版本号 `Z` 从 1 开始依次递增。 | 6.1.1 |
```
Comment (ran-huang) [inline releases/versioning-test.md:31]: @qiancai 做了一个表格，看起来字有点多，你觉得效果怎么样？

File: releases/versioning-test.md [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+
+TiDB 版本的命名方式为 `X.Y.Z`。
+
+- 从 TiDB 1.0 起，`X` 每年依次递增并引入新的功能和改进。
+- `Y` 从 0 开始依次递增，Y 的递增代表引入新的功能和改进。`X.Y` 代表一个版本系列。
+- `Z` 代表补丁版本的编号，一个版本系列第一次发版时 `Z` 默认为 0，补丁版本的 `Z` 从 1 开始依次递增。
+
+| 版本系列 | 定义 | 命名方式 | 示例版本 |
+|:--|:-----|:---|:--|
+| 长期支持版本 (Long-Term Support Releases, LTS) | 长期支持版本约每六个月发布一次，会引入新的功能和改进，也会包含缺陷修复和安全漏洞修复，并会按需在版本生命周期内发布基于当前版本的补丁版本 (Patch Release)。 | 长期支持版本命名方式为 `X.Y`，系列版本号使用一个新的系列，`Z` 默认为 0。 | 6.1.0<br/>5.4.0 |
```
Comment (qiancai) [inline releases/versioning-test.md:29]:
```suggestion
| 长期支持版本 (Long-Term Support Releases, LTS) |  LTS 约每六个月发布一次，会引入新的功能、改进和修复，在生命周期内会按需发布基于当前版本的补丁版本 (Patch Release)。 |  `X.Y.Z`，`Z` 默认为 0。 | 6.1.0<br/>5.4.0 |
```

File: releases/versioning-test.md (hunk 8 above)
Comment (qiancai) [reply releases/versioning-test.md:31]: 的确有点多，或许可以简化下文字

**Review by Ivy-YinSu — APPROVED** [review]

File: releases/versioning-test.md [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+
+| 版本系列 | 定义 | 命名方式 | 示例版本 |
+|:--|:-----|:---|:--|
+| 长期支持版本 (Long-Term Support Releases, LTS) |  LTS 约每六个月发布一次，会引入新的功能、改进和修复，在生命周期内会按需发布基于当前版本的补丁版本 (Patch Release)。 |  `X.Y.Z`，`Z` 默认为 0。 | 6.1.0<br/>5.4.0 |
+| 开发里程碑版本 (Development Milestone Releases, DMR) | DMR 约每两个月发布一次。如遇 LTS 发版，DMR 版本发版时间顺延两个月。DMR 会引入新的功能、改进和修复。但 TiDB 不提供基于 DMR 版本的补丁版本，如有相关 bug 会在后续系列版本中陆续修复。 |  `X.Y.Z`，`Z` 默认为0，并添加后缀 `-DMR`。| 6.0.0-DMR |
+| 补丁版本 (Patch Release) | 补丁版本是在长期支持版本 (LTS) 生命周期内按需发布的补丁版本。补丁版本主要包含 bug 修复、安全漏洞修复，不会包含新的功能。 | 补丁版本命名方式为 `X.Y.Z`。其中，系列版本号 `X.Y` 与对应的长期支持版本保持一致，补丁版本号 `Z` 从 1 开始依次递增。 | 6.1.1 |
+
+## TiDB 工具版本
+
+TiDB 工具 与 TiDB server 分开发布，并使用独立的版本号体系。
```
Comment (lilin90) [inline releases/versioning-test.md:35]:
```suggestion
TiDB 工具与 TiDB server 分开发布，并使用独立的版本号体系。
```

File: releases/versioning-test.md [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+TiDB 版本的命名方式为 `X.Y.Z`。
+
+- 从 TiDB 1.0 起，`X` 每年依次递增并引入新的功能和改进。
+- `Y` 从 0 开始依次递增，Y 的递增代表引入新的功能和改进。`X.Y` 代表一个版本系列。
+- `Z` 代表补丁版本的编号，一个版本系列第一次发版时 `Z` 默认为 0，补丁版本的 `Z` 从 1 开始依次递增。
+
+| 版本系列 | 定义 | 命名方式 | 示例版本 |
+|:--|:-----|:---|:--|
+| 长期支持版本 (Long-Term Support Releases, LTS) |  LTS 约每六个月发布一次，会引入新的功能、改进和修复，在生命周期内会按需发布基于当前版本的补丁版本 (Patch Release)。 |  `X.Y.Z`，`Z` 默认为 0。 | 6.1.0<br/>5.4.0 |
+| 开发里程碑版本 (Development Milestone Releases, DMR) | DMR 约每两个月发布一次。如遇 LTS 发版，DMR 版本发版时间顺延两个月。DMR 会引入新的功能、改进和修复。但 TiDB 不提供基于 DMR 版本的补丁版本，如有相关 bug 会在后续系列版本中陆续修复。 |  `X.Y.Z`，`Z` 默认为0，并添加后缀 `-DMR`。| 6.0.0-DMR |
```
Comment (lilin90) [inline releases/versioning-test.md:30]:
```suggestion
| 开发里程碑版本 (Development Milestone Releases, DMR) | DMR 约每两个月发布一次。如遇 LTS 发版，DMR 版本发版时间顺延两个月。DMR 会引入新的功能、改进和修复。但 TiDB 不提供基于 DMR 版本的补丁版本，如有相关 bug 会在后续系列版本中陆续修复。 |  `X.Y.Z`，`Z` 默认为 0，并添加后缀 `-DMR`。| 6.0.0-DMR |
```

File: releases/versioning-test.md [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+| 补丁版本 (Patch Release) | 补丁版本是在长期支持版本 (LTS) 生命周期内按需发布的补丁版本。补丁版本主要包含 bug 修复、安全漏洞修复，不会包含新的功能。 | 补丁版本命名方式为 `X.Y.Z`。其中，系列版本号 `X.Y` 与对应的长期支持版本保持一致，补丁版本号 `Z` 从 1 开始依次递增。 | 6.1.1 |
```
Comment (lilin90) [inline releases/versioning-test.md:31]:
```suggestion
| 补丁版本 (Patch Release) | 补丁版本是在长期支持版本 (LTS) 的生命周期内按需发布的补丁版本。补丁版本主要包含 bug 修复、安全漏洞修复，不会包含新的功能。 | 补丁版本命名方式为 `X.Y.Z`。其中，系列版本号 `X.Y` 与对应的长期支持版本保持一致，补丁版本号 `Z` 从 1 开始依次递增。 | 6.1.1 |
```

File: releases/versioning-test.md (hunk 10 above)
Comment (lilin90) [reply releases/versioning-test.md:35]: 这里是否要明确一下“TiDB 工具”具体指哪些呢？有些已经跟 TiDB server 发版一致了

File: releases/versioning.md [hunk 13]
```diff
@@ -0,0 +1,100 @@
...
+
+示例版本：
+
+- 1.1 Beta
+- 2.1 Beta
+- 4.0.0-beta.1
+
+### 内部测试版本
+
+内部测试版本 (Alpha Releases) 是内部测试版，会引入新的功能和改进。Alpha 版是当前系列版本的最初版本。Alpha 版可能存在一些 Bug，提供给尝鲜用户，可以用于测试最新的功能。
```
Comment (lilin90) [inline releases/versioning.md:96]: 与文中其它地方出现的大小写保持一致
```suggestion
内部测试版本 (Alpha Releases) 是内部测试版，会引入新的功能和改进。Alpha 版是当前系列版本的最初版本。Alpha 版可能存在一些 bug，提供给尝鲜用户，可以用于测试最新的功能。
```

File: releases/versioning.md [hunk 14]
```diff
@@ -0,0 +1,100 @@
...
+
+示例版本：
+
+- RC1
+- 2.0-RC1
+- 3.0.0-rc.1
+
+### 公开测试版本
+
+公开测试版本 (Beta Releases) 会引入新的功能和改进，相对于内部测试版本已有了很大的改进，消除了严重的错误，但还是存在着一些缺陷，提供给尝鲜用户，可以用于测试最新的功能。
```
Comment (lilin90) [inline releases/versioning.md:86]: 这里的“缺陷”是不是也指 bug？如果是的话建议统一一下

**Review by lilin90 — APPROVED** [review]
LGTM

**Pull request merged**